apps/studio/components/interfaces/Auth/EmailTemplates/TemplateEditor.tsx

```diff
@@ -75,9 +74,13 @@ export const TemplateEditor = ({ template }: TemplateEditorProps) => {
   const [bodyValue, setBodyValue] = useState((authConfig && authConfig[messageSlug]) ?? '')
   const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false)
   const [isSavingTemplate, setIsSavingTemplate] = useState(false)
+  const [activeView, setActiveView] = useState<'source' | 'preview'>('source')
+  const [lastAttemptedSaveValues, setLastAttemptedSaveValues] = useState<{
+    formValues: { [key: string]: string }
+    bodyValue: string
+  } | null>(null)
```

Comment:
This approach to enabled/disabling the _Save changes_ button is necessary, as far as I can see. Otherwise when spam is detected, the save is blocked but `hasChanges` remains true because the values weren’t ever saved.

The currently-in-prod approach is to set `preventSaveFromSpamCheck` in the disabled condition. I believe that depends on the current `validationResult`, which can change, making the button state unreliable.

So this is an attempt at a simpler fix: track a flag indicating a spam-blocked save attempt, and using that in the disabled condition.

Reply:
hmm one sec - i don't think we need to save this value

will look into simplifying this, the save being blocked was probably the issue (although i can't recall what it was before)